websites/L/Letterboxd/presence.ts

```diff
@@ -0,0 +1,543 @@
+							presenceData.largeImageKey = getImageURLByAlt(
+								clarifyString(title.innerText)
+							);
+							presenceData.smallImageKey = "final";
+
+							break;
+						}
+
+						case "film": {
+							const title = clarifyString(
+									(
+										document.getElementsByClassName("film-title-wrapper")[0]
+											.firstElementChild as HTMLAnchorElement
+									).innerText
+								),
+								rater = filterIterable(
+									document.getElementsByTagName("span"),
+									val => {
+										if (val.getAttribute("itemprop") === "name") return true;
+									}
+								).innerText,
+								rating = filterIterable(
+									document.getElementsByTagName("span"),
+									val => {
+										if (val.className.startsWith("rating rating-large"))
+											return true;
+									}
+								).innerText;
+
+							presenceData.details = `Review of ${title}`;
+							presenceData.state = `By ${rater} (${rating})`;
+							presenceData.buttons = [
+								{ label: "View review", url: window.location.href }
+							];
+							presenceData.largeImageKey = getImageURLByAlt(title);
+							presenceData.smallImageKey = getImageURLByAlt(rater);
+							presenceData.smallImageText = rater;
+
+							break;
+						}
+					}
+
+					if (
+						[
+							"watchlist",
+							"films",
+							"activity",
+							"blocked",
+							"followers",
+							"following",
+							"tags",
+							"likes",
+							"lists"
+						].includes(path[1])
+					) {
+						const name = (
+								document.getElementsByClassName("title-3")[0]
+									.firstElementChild as HTMLAnchorElement
+							).innerText,
+							smallPFP = getImageURLByAlt(name);
+
+						if (path[0] !== user && path[0] !== user.toLowerCase()) {
+							presenceData.details = presenceData.details
+								.replace("their", `${name}'s`)
+								.replace("they've", `${name} has`);
+						}
+
+						presenceData.smallImageKey = smallPFP;
+						presenceData.smallImageText = name;
+					}
+				} else {
+					const name = (
+						document.getElementsByClassName("title-1")[0] as HTMLHeadingElement
+					).innerText;
+					// I could use the get image func but ID is available so its fine (smaller loop (I think))
+					presenceData.details = `Viewing ${
+						path[0] === user ? "their own" : `${name}'s`
+					} profile`;
+					presenceData.state = `(${
+						path[0] === user ? `${name}/${path[0]}` : path[0]
+					})`;
+					presenceData.largeImageKey = (
+						document.getElementById("avatar-zoom")
+							.previousElementSibling as HTMLImageElement
+					).src;
+					presenceData.smallImageKey = "final";
+					presenceData.buttons = [
+						{ label: `View ${name}`, url: window.location.href }
+					];
+
+					break;
+				}
+		}
+	} else presenceData.details = "At home";
+
+	if (!(await presence.getSetting("show_buttons")))
+		// eslint-disable-next-line no-undefined
+		presenceData.buttons = undefined;
+
+	if (details !== presenceData.details) {
```

Comment:
Pretty unnecessary

Reply:
Its there for not messing up the `Date.now()` of "Time elapsed"